metadata-ingestion/src/datahub/ingestion/source/data_lake_common/path_spec.py

```diff
@@ -81,6 +81,30 @@ def allowed(self, path: str) -> bool:
         logger.debug(f"{path} allowed for dataset creation")
         return True
 
+    def dir_allowed(self, path: str) -> bool:
+        path_slash = path.count("/")
+        glob_slash = self.glob_include.count("/")
+        if path_slash > glob_slash:
+            return False
+        slash_to_remove_from_glob = glob_slash - path_slash
+
+        glob_include = self.glob_include.rsplit("/", 1)[0]
+
+        for i in range(slash_to_remove_from_glob):
+            glob_include = glob_include.rsplit("/", 1)[0]
```

Comment:
Could we get this via `glob_include.rsplit("/", i)[0]`, so I guess `glob_include.rsplit("/", slash_to_remove_from_glob + 1)[0]` in total? If that's the first time it's defined

Reply:
Thanks, I made it more clear in the code.